addons/udes_security/controllers/main.py

```diff
@@ -87,17 +96,36 @@ def _get_menu_info_from_action_id(self, action_id):
         """
         request.env.cr.execute(query, (action_id,))
         result = request.env.cr.fetchone()
-        if not result:
-            return [], []
-        (parent_path, relation_table, inverse, action_id) = result
-        # Get ids in order from parent --> child of the menus
-        parent_path_ids = [int(str_menu_id) for str_menu_id in parent_path.split("/")[:-1]]
-        action_group_ids = []
-        if relation_table and inverse:
-            action_group_ids = self._get_object_group_ids(action_id, relation_table, inverse)
+        if result:
+            (parent_path, relation_table, inverse, action_id) = result
+            # Get ids in order from parent --> child of the menus
+            parent_path_ids = [int(str_menu_id) for str_menu_id in parent_path.split("/")[:-1]]
+            if relation_table and inverse:  # Some ir.actions* models don't have groups_id column
+                action_group_ids = self._get_object_group_ids(action_id, relation_table, inverse)
+        else:
+            # Certain actions don't have a menu, but we still need to determine the model
+            # which the action_id is for to get its groups. This approach is slower.
+            # This will happen rarely, so not concerned about optimising.
+            models = [
+                "ir.actions.act_window",
+                "ir.actions.server",
+                "ir.actions.client",
+                "ir.actions.report",
+                "ir.actions.act_url",
+            ]
+            for model in models:
+                action = request.env[model].search([("id", "=", action_id)])
+                if action:
+                    action_group_ids = action.groups_id.ids
```

Comment:
Q: Can we try disabling prefetching here? I think `groups_id.ids` might go and collect a load of fields we don't need or want.

Reply:
testing menuitem after a restart and re-log in, it appears to be slightly faster with your suggestion
Settings/Technical/Channels
without prefetch change
```action = request.env[model].search([("id", "=", action_id)])```
Hook took 0.008173465728759766
with prefetch change
```action = request.env[model].with_context(prefetch_fields=False).search([("id", "=", action_id)])```
Hook took 0.006898164749145508